docs/ru/query_language/functions/array_functions.md

```diff
@@ -658,4 +658,115 @@ SELECT arrayReverseSort((x, y) -> -y, [4, 3, 5], [1, 2, 3]) AS res;
 
 Особенная функция. Смотрите раздел ["Функция arrayJoin"](array_join.md#functions_arrayjoin).
 
+## arrayDifference(arr)
+
+Принимает массив, возвращает массив разностей между парами соседних элементов. 
+
+Пример:
+
+```sql
+SELECT arrayDifference([1, 2, 3, 4])
+```
+
+```text
+┌─arrayDifference([1, 2, 3, 4])─┐
+│ [0,1,1,1]                     │
+└───────────────────────────────┘
+```
+
+## arrayDistinct(arr)
+
+Принимает массив, возвращает массив, содержащий уникальные элементы. 
+
+Пример:
+
+```sql
+SELECT arrayDistinct([1, 2, 2, 3, 1])
+```
+
+```text
+┌─arrayDistinct([1, 2, 2, 3, 1])─┐
+│ [1,2,3]                        │
+└────────────────────────────────┘
+```
+
+## arrayEnumerateDense(arr)
+
+Возвращает массив того же размера, что и исходный массив, с индексами исходного массива, указывающими, где каждый элемент впервые появляется в исходном массиве.
+
+Пример:
+
+```sql
+SELECT arrayEnumerateDense([10, 20, 10, 30])
+```
+
+```text
+┌─arrayEnumerateDense([10, 20, 10, 30])─┐
+│ [1,2,1,3]                             │
+└───────────────────────────────────────┘
+```
+
+## arrayIntersect(arr)
+
+Принимает массивы, возвращает пересечение элементов массивов.
+
+Пример:
+
+```sql
+SELECT
+    arrayIntersect([1, 2], [1, 3], [2, 3]) AS no_intersect,
+    arrayIntersect([1, 2], [1, 3], [1, 4]) AS intersect
+```
+
+```text
+┌─no_intersect─┬─intersect─┐
+│ []           │ [1]       │
+└──────────────┴───────────┘
+```
+
+## arrayReduce(agg_func, arr1, ...)
+
+Применяет агрегатную функцию к массиву и возвращает ее результат.
```

Comment:
Should note that the aggregate function name is passed as a constant string literal.
And about the way how parametric aggregate functions can be specified.